daemon/clients/__init__.py

```diff
@@ -0,0 +1,109 @@
+from typing import Optional, Dict, TYPE_CHECKING, Union, Awaitable
+
+from jina.logging.logger import JinaLogger
+
+from .base import BaseClient, AsyncBaseClient
+from .peas import PeaClient, AsyncPeaClient
+from .pods import PodClient, AsyncPodClient
+from .flows import FlowClient, AsyncFlowClient
+from .workspaces import WorkspaceClient, AsyncWorkspaceClient
+
+
+__all__ = ['JinaDClient', 'AsyncJinaDClient']
+
+
+if TYPE_CHECKING:
+    from ..models.id import DaemonID
+
+
+class JinaDClient:
+    """JinaD Client
+
+    :param host: hostname of remote JinaD server
+    :param port: port of remote JinaD server
+    :param timeout: default timeout for requests, defaults to None
+    """
+
+    _base_cls = BaseClient
+    _pea_cls = PeaClient
+    _pod_cls = PodClient
+    _flow_cls = FlowClient
+    _workspace_cls = WorkspaceClient
+
+    def __init__(
+        self,
+        host: str,
+        port: int,
+        timeout: Optional[float] = None,
+        logger: JinaLogger = None,
+    ) -> None:
+        uri = f'{host}:{port}'
+        timeout = timeout
+        logger = logger or JinaLogger(self.__class__.__name__)
+        self.kwargs = {'uri': uri, 'logger': logger, 'timeout': timeout}
+
+    @property
+    def peas(self):
+        """Pea Client
+
+        :return: Pea Client
+        """
+        return self._pea_cls(**self.kwargs)
```

Comment:
should we somehow cache all of these individual clients instead of always returning a new instance? Right now it seems cheap enough to create new instances all the time, but maybe it gets more expensive in the future?

Reply:
Hmm, we possibly can. Maybe in the next iteration?